docs/howto/architectures.rst

```diff
@@ -0,0 +1,234 @@
+      build-for: amd64
+
+Building on ``amd64`` will produce one snap built for ``amd64``. Snapcraft will
+raise an error when building on another architecture.
+
+If ``build-for`` is omitted, then it will assume the value of ``build-on``. The
+following snippet snippet will produce the same result:
+
+.. code-block:: yaml
+
+  architectures:
+    - build-on: amd64
+
+The shorthand format will also produce the same result:
+
+.. code-block:: yaml
+
+  architectures:
+    - amd64
+
+core20
+^^^^^^
+
+.. code-block:: yaml
+
+  architectures:
+    - build-on: amd64
+      run-on: amd64
+
+Building on ``amd64`` will produce one snap built for ``amd64``. Snapcraft will
+not raise an error when building on another architecture. Instead, it will
+ignore the ``architectures`` keyword and build for the build-on architecture.
+
+If ``run-on`` is omitted, then it will assume the value of ``build-on``. The
+following snippet snippet will produce the same result:
+
+.. code-block:: yaml
+
+  architectures:
+    - build-on: amd64
+
+The shorthand format will also produce the same result:
+
+.. code-block:: yaml
+
+  architectures:
+    - amd64
+
+How to create a set of snaps for multiple architectures
+-------------------------------------------------------
+
+core22
+^^^^^^
+
+.. code-block:: yaml
+
+  architectures:
+    - build-on: amd64
+      build-for: amd64
+    - build-on: arm64
+      build-for: arm64
+
+Building on ``amd64`` will produce one snap for ``amd64``. Building on ``arm64``
+will produce one snap for ``arm64``. Snapcraft will raise an error when building
+on another architecture.
+
+If ``build-for`` is omitted, then it will assume the value of ``build-on``. The
+following snippet snippet will produce the same result:
+
+.. code-block:: yaml
+
+  architectures:
+    - build-on: amd64
+    - build-on: arm64
+
+The shorthand format will also produce the same result:
+
+.. code-block:: yaml
+
+  architectures:
+    - amd64
+    - arm64
+
+core20
+^^^^^^
+
+.. code-block:: yaml
+
+  architectures:
+    - build-on: amd64
+      run-on: amd64
+    - build-on: arm64
+      run-on: arm64
+
+Building on ``amd64`` will produce one snap built for ``amd64``. Building on
+``arm64`` will produce one snap built for ``arm64``. Snapcraft will not raise
+an error when building on another architecture. Instead, it will ignore the
+``architectures`` keyword and build for the build-on architecture.
+
+If ``run-on`` is omitted, then it will assume the value of ``build-on``. The
+following snippet snippet will produce the same result:
+
+.. code-block:: yaml
+
+  architectures:
+    - build-on: amd64
+    - build-on: arm64
+
+The shorthand format will also produce the same result:
+
+.. code-block:: yaml
+
+  architectures:
+    - amd64
+    - arm64
+
+How to create an architecture independent snap
+----------------------------------------------
+
+core22
+^^^^^^
+
+.. code-block:: yaml
+
+  architectures:
+    - build-on: amd64
+      build-for: all
```

Comment:
Is there a `build-on: all`?

Reply:
Only `build-for: [all]`.  I added more details about that in the reference section and a link to the reference section.